feat(react-grid): implement Inline Cell Editing (#2302)
DevExpress/devextreme-reactive: packages/dx-react-grid/docs/reference/table-inline-cell-editing.md

editingEnabled | boolean | Specifies whether editing a column is enabled.
onValueChange | (newValue: any) => void | Handles value changes.
autoFocus | boolean | Define autoFocus on cell when editing start (`true` by default).
onFocus | boolean | Define actions when cell get focus.
onBlur | () => void | Define actions when cell lost focus.
onKeyDown | (key: string) => void | Define actions when keyboard key pressed.

## Plugin Components

Name | Properties | Description
-----|------------|------------
TableInlineCellEditing.Cell | [TableInlineCellEditing.CellProps](#tableinlinecelleditingcellprops) | A component that renders an editable cell.

Additional properties are added to the component's root element.

## Plugin Developer Reference

### Imports

Name | Plugin | Type | Description
-----|--------|------|------------
editingCells | [Getter](../../../dx-react-core/docs/reference/getter.md) | Array&lt;{rowId: number &#124; string, columnName: string}&gt; | Row ID and column name of cells that are being edited.
getCellValue | [Getter](../../../dx-react-core/docs/reference/getter.md) | (row: any, columnName: string) => any | A function used to get the specified row's column value.

> createRowChange | [Getter](../../../dx-react-core/docs/reference/getter.md) | (row: any, value: any, columnName: string) => any | A function that returns a value that specifies row changes depending on the row's editable cell values. This function is called each time an editor's value changes.

Reviewer: ```suggestion
createRowChange | [Getter](../../../dx-react-core/docs/reference/getter.md) | (row: any, value: any, columnName: string) => any | A function that allows you to modify the edited cell value before it is added to the `rowChanges` array. This function is called each time an editor's value changes and should return the modified cell value.
```

> rowChanges | [Getter](../../../dx-react-core/docs/reference/getter.md) | { [key: string]: any } | An associative array that stores changes made to existing rows. Each array item specifies changes made to a row. The item's key specifies the associated row's ID.

Reviewer: ```suggestion
rowChanges | [Getter](../../../dx-react-core/docs/reference/getter.md) | { [key: string]: any } | An associative array in which each item consists of a row ID (the `key` field) and changes made to this row.
```

> isColumnEditingEnabled | [Getter](../../../dx-react-core/docs/reference/getter.md) | (columnName: string) => boolean |  A function that returns a value that specifies if editing by a column is enabled.

Reviewer: ```suggestion
isColumnEditingEnabled | [Getter](../../../dx-react-core/docs/reference/getter.md) | (columnName: string) => boolean |  A function that returns a Boolean value that specifies whether users are allowed to edit the column identified by the `columnName`.
```

> changeRow | [Action](../../../dx-react-core/docs/reference/action.md) | ({ rowId: number &#124; string, change: object }) => void | Applies a change to an existing row.

Reviewer: ```suggestion
changeRow | [Action](../../../dx-react-core/docs/reference/action.md) | ({ rowId: number &#124; string, change: object }) => void | Applies a change to a row identified by ID.
```

> startEditCells | [Action](../../../dx-react-core/docs/reference/action.md) | ({ editingCells: Array&lt;{rowId: number &#124; string, columnName: string}&gt; }) => void | Switches cells with the specified row ID and column name to the edit mode.

Reviewer: ```suggestion
startEditCells | [Action](../../../dx-react-core/docs/reference/action.md) | ({ editingCells: Array&lt;{rowId: number &#124; string, columnName: string}&gt; }) => void | Switches cells identified by the row ID and column name to edit mode.
```

> stopEditCells | [Action](../../../dx-react-core/docs/reference/action.md) | ({ editingCells: Array&lt;{rowId: number &#124; string, columnName: string}&gt; }) => void | Switches cells with the specified row ID and column name to the read-only mode.

Reviewer: ```suggestion
stopEditCells | [Action](../../../dx-react-core/docs/reference/action.md) | ({ editingCells: Array&lt;{rowId: number &#124; string, columnName: string}&gt; }) => void | Switches cells identified by the row ID and column name to read-only mode.
```

> cancelChangedRows | [Action](../../../dx-react-core/docs/reference/action.md) | ({ rowIds: Array&lt;number &#124; string&gt; }) => void | Cancels uncommitted changes in rows with the specified ID.

Reviewer: ```suggestion
cancelChangedRows | [Action](../../../dx-react-core/docs/reference/action.md) | ({ rowIds: Array&lt;number &#124; string&gt; }) => void | Cancels uncommitted changes in rows with the specified IDs.
```

> commitChangedRows | [Action](../../../dx-react-core/docs/reference/action.md) | ({ rowIds: Array&lt;number &#124; string&gt; }) => void | Fires the `onCommitChanges` event with the corresponding [ChangeSet](editing-state.md#changeset) and removes specified rows from the `rowChanges` array.

Reviewer: ```suggestion
commitChangedRows | [Action](../../../dx-react-core/docs/reference/action.md) | ({ rowIds: Array&lt;number &#124; string&gt; }) => void | Raises the `onCommitChanges` event with the corresponding [ChangeSet](editing-state.md#changeset) and removes the specified rows from the `rowChanges` array.
```
